cocos/2d/CCCamera.cpp

```diff
@@ -361,4 +365,76 @@ void Camera::setScene(Scene* scene)
     }
 }
 
+void Camera::clearBackground(float depth)
+{
+    GLboolean oldDepthTest;
+    GLint oldDepthFunc;
+    GLboolean oldDepthMask;
+    {
+        glColorMask(GL_FALSE, GL_FALSE, GL_FALSE, GL_FALSE);
+        glStencilMask(0);
+        oldDepthTest = glIsEnabled(GL_DEPTH_TEST);
+        glGetIntegerv(GL_DEPTH_FUNC, &oldDepthFunc);
+        glGetBooleanv(GL_DEPTH_WRITEMASK, &oldDepthMask);
+        glDepthMask(GL_TRUE);
+        glEnable(GL_DEPTH_TEST);
+        glDepthFunc(GL_ALWAYS);
+    }
+    
+    //draw
+    static V3F_C4B_T2F_Quad quad;
+    quad.bl.vertices = Vec3(-1,-1,0);
+    quad.br.vertices = Vec3(1,-1,0);
+    quad.tl.vertices = Vec3(-1,1,0);
+    quad.tr.vertices = Vec3(1,1,0);
+    
+    quad.bl.colors = quad.br.colors = quad.tl.colors = quad.tr.colors = Color4B(0,0,0,1);
+    
+    quad.bl.texCoords = Tex2F(0,0);
+    quad.br.texCoords = Tex2F(1,0);
+    quad.tl.texCoords = Tex2F(0,1);
+    quad.tr.texCoords = Tex2F(1,1);
+    
+    auto shader = GLProgramCache::getInstance()->getGLProgram(GLProgram::SHADER_CAMERA_CLEAR);
+    auto programState = GLProgramState::create(shader);
```

Comment:
Does the glClear work for your purpose? It should run fast than this.

Also GLProgramState::getOrCreateWithGLProgram(shader) can avoid to create GLProgramState every frame in your case.